docker_cli/image.py

```diff
@@ -0,0 +1,458 @@
+
+        Raises
+        -------
+        CalledProcessError
+            If the docker inspect command fails.
+        """
+        cmd = ["docker", "inspect"]
+        if format:
+            cmd += [f"-f={format}"]
+        cmd += [self._id]
+
+        inspect_result = run(
+            cmd,
+            capture_output=True,
+            text=True,
+            check=True
+        )
+
+        output_text = inspect_result.stdout
+        return output_text
+
+    def run(
+        self,
+        command: str,
+        *,
+        stdout: Optional[Union[io.TextIOBase, int]] = None,
+        stderr: Optional[Union[io.TextIOBase, int]] = None,
+        interactive: bool = False,
+        network: str = "host"
+    ):
+        """
+        Run the given command on a container.
+
+            .. note::
+        For stderr special values to pass into `stdout` and `stdin`, review
+        values passable into the same arguments in the :func:`~subprocess.run`
+        function:
+        https://docs.python.org/3/library/subprocess.html#subprocess.run
+
+            .. warning::
+        This method does not work correctly if the image built does not have
+        bash installed.
+
+        Parameters
+        ----------
+        cmd : str
+            The desired command, in linux command format.
+        stdout : io.TextIOBase or subprocess special value or None, optional
+            The location to send the process stdout output to.Defaults to None.
+        stderr : io.TextIOBase or subprocess special value or None, optional
+            The location to send the process stderr output to.
+            Defaults to None.
+        interactive : bool, optional
+            A boolean describing whether or not to run this command in
+            interactive mode or not. Defaults to True.
+        network : str, optional
+            The name of the network. Defaults to "host".
+
+        Returns
+        -------
+        str, optional
+            The output of the process, if `stdout` == PIPE.
+
+        Raises
+        -------
+        CalledProcessError
+            If the docker inspect command fails.
+        CommandNotFoundOnImageError:
+            When a command is attempted that is not recognized on the image.
+        """
+        cmd = ["docker", "run", f"--network={network}", "--rm"]
+        if interactive:
+            cmd += ["-i"]
+            if stdin.isatty():
+                cmd += ["--tty"]
+        cmd += [self._id, "bash"]
+        cmd += ["-ci"] if interactive else ["-c"]
+        cmd += split(f"'{command}'")
+
+        try:
+            result = run(
+                cmd,
+                text=True,
+                stdout=stdout,  # type: ignore
+                stderr=stderr,  # type: ignore
+                check=True
+            )
+            retval = result.stdout
+        except CalledProcessError as err:
+            if err.returncode == 127:
+                raise CommandNotFoundOnImageError(err, split(command)[0])
```

Comment:
Let's use [exception chaining](https://peps.python.org/pep-3134/#explicit-exception-chaining) here:

```suggestion
                raise CommandNotFoundOnImageError(err, split(command)[0]) from err
```

Using this syntax, if an exception is encountered here, the user will see both the `CommandNotFoundOnImageError` exception and the original exception that caused it. They'll also be able to see the full traceback history of both exceptions.